tinytuya/core.py

```diff
@@ -648,28 +650,52 @@ def device_info( dev_id ):
             "command": {"devId": "", "uid": "", "t": ""},
         },
     },
+    # v3.4+ devices do not need devId/gwId/uid
     "v3.4": {
         CONTROL: {
             "command_override": CONTROL_NEW,  # Uses CONTROL_NEW command
-            "command": {"protocol":5, "t": "int", "data": ""}
+            "command": {"protocol":4, "t": "int", "data": {}}
             },
-        DP_QUERY: { "command_override": DP_QUERY_NEW },
+        CONTROL_NEW: {
+            "command": {"protocol":4, "t": "int", "data": {}}
+        },
+        DP_QUERY: {
+            "command_override": DP_QUERY_NEW,
+            "command": {"protocol":4, "t": "int", "data": {}}
+        },
+        DP_QUERY_NEW: {
+            "command": {"protocol":4, "t": "int", "data": {}}
+        },
     },
+    # v3.5 is just a copy of v3.4
     "v3.5": {
         CONTROL: {
             "command_override": CONTROL_NEW,  # Uses CONTROL_NEW command
-            "command": {"protocol":5, "t": "int", "data": ""}
+            "command": {"protocol":4, "t": "int", "data": {}}

```

Comment:
Seem very odd, no?  I saw your note but why use protocol 4 for 3.4 and 3.5?  Either we haven't figure out there scheme or Tuya has a loose/confusing approach to versioning. 😂

Reply:
It's most likely the latter.  In my packet captures the Smart Life app was sending protocol:5 while the v3.4 device was responding with protocol:4.

Request: `{"data":{"cid":"tu####4","ctype":0,"dps":{"2":false}},"protocol":5,"t":1679183236}'`
Response: {"protocol":4,"t":1679183236,"data":{"dps":{"2":false},"cid":"tu####4","type":"query"}}'